blockstore/splitstore/splitstore_compact.go

```diff
@@ -1372,9 +1372,21 @@ func (s *SplitStore) purge(coldr *ColdSetReader, checkpoint *Checkpoint, markSet
 		return err
 	}
 
+	now := time.Now()
+
 	err := coldr.ForEach(func(c cid.Cid) error {
 		batch = append(batch, c)
 		if len(batch) == batchSize {
+			// add some time slicing to the purge as this a very disk I/O heavy operation that
+			// requires write access to txnLk that may starve other operations that require
+			// access to the blockstore.
+			elapsed := time.Since(now)
+			if elapsed > time.Second {
+				// work 1 second, sleep 4, or 20% utilization
+				time.Sleep(4 * elapsed)
```

Comment:
`4 * elapsed`? Do we not want `4 * time.Second` here?

Reply:
if we schedule slightly over second, this means that we sleep a bit longer guaranteeing a 20% utilization

Reply:
Yup, that makes sense.